services/application.js

```diff
@@ -143,6 +143,27 @@ class Application {
         });
         return (updated.modifiedCount && updated.modifiedCount > 0) ? await this.dbService.find(APPLICATION, {_id: document._id}) : null;
     }
+
+    async deleteInactiveApplications(inactiveDays) {
+        const inactiveCondition = {
+            updatedAt: {
+                $lt: subtractDaysFromNow(inactiveDays)
+            },
+            status: SUBMITTED
+        };
+        const applications = await this.applicationCollection.aggregate([{$match: inactiveCondition}, {"$limit": 1}]);
+        if (applications.length > 0) {
```

Comment:
let's make this check null safe and throw an error message if the search fails and "applications" is null:
`if (!applications) throw new Error("error message here")`
`if (applications?.length > 0) {`
`...`

Reply:
Updated.